### ISSUE TITLE
API key required

### ISSUE THREAD

**Issue by diachedelic** [NARKOZ/mmc#4]
Running `mmc 1 AUD to USD` I get the error message `No results for currency rate AUD_USD`.

Looking at the source code, if I run `curl http://free.currencyconverterapi.com/api/v3/convert?q=USD_AUD` I get `{"status":400,"error":"API Key is required."}`

**Comment by NARKOZ** [comment]
Yep, API key is now required for the free server. Time to look for another free currency API.

**Comment by diachedelic** [comment]
The API key is apparently free, perhaps there could be a way of providing one to mmc?

**Comment by NARKOZ** [comment]
> The API key is apparently free, perhaps there could be a way of providing one to mmc?

That's also an option. However, I'd prefer it to work out of the box.

**Comment by diachedelic** [comment]
Or switch to OpenRates, no API key required apparently: https://davidwalsh.name/openrates <https://davidwalsh.name/openrates>

**Comment by diachedelic** [comment]
You could bundle an API key...

**Comment by NARKOZ** [comment]
> Or switch to OpenRates, no API key required apparently: https://davidwalsh.name/openrates

Not all currencies are supported by OpenRates https://api.exchangeratesapi.io/latest

**Comment by diachedelic** [comment]
pr submitted, I have never written in go before so please let me know if I've done something ugly

**Comment by diachedelic** [comment]
oh whoops I didnt see your comment - well...some currencies are better than nothing

**Comment by LuanPetruitis** [comment]
you can enter this site https://free.currencyconverterapi.com/   And click on get your free API key. you put your email. They will send an email with the key
